src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -368,7 +380,44 @@ var ReactDOM = {
       // allows arrays.
       invariant(
         isValidElement(element),
-        'render(): Invalid component element.'
+        'ReactDOM.render(): Invalid component element.%s',
+        (
+          typeof element === 'string' ?
+            ' Instead of passing a string like \'div\', pass ' +
+            'React.createElement(\'div\') or <div />.' :
+          typeof element === 'function' ?
+            ' Instead of passing a class like Foo, pass ' +
+            'React.createElement(Foo) or <Foo />.' :
+          // Check if it quacks like an element
+          element != null && element.props !== undefined ?
+            ' This may be caused by unintentionally loading two independent ' +
+            'copies of React.' :
+            ''
+        )
```

Comment:
It would be nice if error code transform was throwing on anything other than static messages.